### ISSUE TITLE
content_for :html_head - in 1.2.0 - method not found

### ISSUE THREAD

**Issue by crh3675** [browsermedia/bcms_news#2]
app/views/portlets/recent_news/render.html.erb

```
<% content_for :html_head do %>
    <%= auto_discovery_link_tag( :rss, news_articles_url, {:title => "RSS Feed for News Articles" }) %>
<% end %>
```

Rails 3 throws method not defined error.  Also, 

```
auto_discovery_link_tag( :rss, news_articles_url, :title => "RSS Feed for News Articles" )
```

syntax is no longer valid.  Needs to be updated to:

```
auto_discovery_link_tag( :rss, news_articles_url, {:title => "RSS Feed for News Articles" })
```

**Comment by Rah1x** [comment]
Well, the issue occurs because somehow in the new version combination (BCMS 3.3, Rails3, Ruby 1.9), the  render.html.erb is called before the the content_for method is called. Here is a snapshot of the log:

``` .
.
.
.
Error occurred while rendering RecentNewsPortlet#2: undefined method `content_for' for nil:NilClass
.
.
/browsercms-3.3.1/app/models/cms/view_context.rb:41:in `content_for'
.
```

**Comment by Rah1x** [comment]
As you can see, "view_context.rb" is called just after the rendering of the erb file (which is called from "ruby\gems\1.9.1\gems\browsercms-3.3.1\app\models\cms\view_context.rb")

Anyway, since this prevents the page from opening, have found the solution to prevent the error (Note: this will prevent the RSS as well - until I figure out how to solve it completely).

In the file: gems\1.9.1\gems\bcms_news-1.2.1\app\views\portlets\news_archive\render.html.erb at the top:

`````` <%
begin
    content_for :html_head do
    %>
    <%= auto_discovery_link_tag( :rss, news_articles_url, {:title => "RSS Feed for News Articles"}) %>
    <%
    end
rescue
end
%>```

``````

**Comment by peakpg** [comment]
This is fixed and will be released in v1.2.3. I have removed the autodiscovery link from the portlets. Autodiscovery needs to be added to page templates (since portlets can't add things via content_for) and I have updated the README accordingly.
